frontend/probloom/src/containers/ProblemSet/ProblemSetDetail/ProblemSetDetail.tsx

```diff
@@ -1,16 +1,263 @@
+    this.props.history.push(
+      '/problem/' + this.props.match.params.id + '/explanation/'
+    );
+  };
+
+  onClickEditCommentButton = (comment: Comment) => {
+    this.setState({ isEdit: true, editComment: comment });
+  };
+
+  onClickDeleteCommentButton = (comment: Comment) => {
+    this.props.onDeleteComment(comment.id);
+    this.props.onGetCommentsOfProblemSet(parseInt(this.props.match.params.id));
+    this.setState({ commentContent: '', isEdit: false, editComment: null });
+  };
+
+  onClickCommentButton = () => {
+    if (this.state.isEdit) {
+      const comment = {
+        id: this.state.editComment?.id,
+        content: this.state.commentContent,
+      };
+      this.props.onUpdateComment(comment);
+      this.setState({ commentContent: '', isEdit: false, editComment: null });
+    } else {
+      const comment = {
+        userID: this.props.selectedUser.id,
+        username: this.props.selectedUser.username,
+        problemSetID: parseInt(this.props.match.params.id),
+        content: this.state.commentContent,
+      };
+      this.props.onCreateComment(comment);
+      this.setState({ commentContent: '' });
+    }
+    this.props.onGetCommentsOfProblemSet(parseInt(this.props.match.params.id));
+  };
+
   render() {
+    if (this.props.selectedUser === null) {
+      this.props.history.push('/signin');
+    }
+    if (this.props.selectedUser) {
+      if (this.props.selectedUser.logged_in === false) {
+        this.props.history.push('/signin');
+      }
+    }
+
+    let isCreator = false;
+    let isSolver = false;
+    if (this.props.selectedProblemSet) {
+      isCreator =
+        this.props.selectedProblemSet.userID === this.props.selectedUser.id;
+      const solver = this.props.solvers.find(
+        (element: Solver) => element.userID === this.props.selectedUser.id
+      );
+      isSolver = solver === undefined;
+    } else {
+      return <NotFound />;
+    }
+
     return (
       <div className="ProblemSetDetail">
-        <div className="title">
-          <h1>ProblemSetDetail Page</h1>
+        <ProblemSetView
+          isCreator={isCreator}
+          isSolver={isSolver}
+          title={this.props.selectedProblemSet.title}
+          content={this.props.selectedProblemSet.content}
+          onClickSignOutButton={() => this.onClickSignOutButton()}
+          onClickBackButton={() => this.onClickBackButton()}
+          onClickEditProblemButton={() => this.onClickEditProblemButton()}
+          onClickDeleteProblemButton={() => this.onClickDeleteProblemButton()}
+          onClickSolveProblemButton={() => this.onClickSolveProblemButton()}
+          onClickExplanationButton={() => this.onClickExplanationButton()}
+        />
+        <div className="Comment">
+          <label>Comment</label>
+          {this.props.comments.map((com, index) => {
+            isCreator = com.userID === this.props.selectedUser.id;
+            return (
+              <CommentCompoent
+                key={index}
```

Comment:
Can we use `com.id` instead of `index` for `key` prop?

Reply:
I'll try it